pandas/_libs/algos.pyx

```diff
@@ -1099,43 +1101,35 @@ def rank_2d(
     argsorted = _as.astype('i8')
 
     for i in range(n):
-        if rank_t is object:
-            dups = sum_ranks = infs = 0
-        else:
-            dups = sum_ranks = 0
+        dups = sum_ranks = infs = 0
 
         total_tie_count = 0
         count = 0.0
         for j in range(k):
-            if rank_t is not object:
-                sum_ranks += j + 1
-                dups += 1
-
             val = values[i, j]
 
-            if rank_t is not uint64_t:
```

Comment:
i expect we'll see a perf hit on non-float dtypes because of mask[...] lookups.  is that effect small?

Reply:
Thanks for bringing this up - good lesson for me to always check perf. Turns out `mask` wasn't typed, so perf got way worse. With mask typed, looks like a small effect. Using the added benchmark from #38744:
```
       before           after         ratio
     [e85d0782]       [36884e67]
     <master>         <bug/2d_rank_inf>
       6.66±0.2ms       6.61±0.2ms     0.99  frame_methods.Rank.time_rank('float')
       2.24±0.1ms      2.58±0.09ms    ~1.15  frame_methods.Rank.time_rank('int')
       46.6±0.5ms         46.6±1ms     1.00  frame_methods.Rank.time_rank('object')
      2.20±0.08ms      2.20±0.06ms     1.00  frame_methods.Rank.time_rank('uint')
```
On a larger (probably more stable example):
```
             before           after         ratio
     [e85d0782]       [36884e67]
     <master>         <bug/2d_rank_inf>
         975±20ms         972±30ms     1.00  frame_methods.Rank.time_rank('float')
+        329±10ms         378±10ms     1.15  frame_methods.Rank.time_rank('int')
       12.8±0.01s        13.4±0.3s     1.05  frame_methods.Rank.time_rank('object')
          318±6ms         343±30ms     1.08  frame_methods.Rank.time_rank('uint')
 ```

Reply:
Also changed if conditions to short-circuit to avoid some mask lookups